packages/amplify-graphql-transformer-core/src/transformation/utils.ts

```diff
@@ -0,0 +1,180 @@
+import {
+  DirectiveNode,
+  TypeDefinitionNode,
+  FieldDefinitionNode,
+  InputValueDefinitionNode,
+  EnumValueDefinitionNode,
+  DirectiveDefinitionNode,
+  TypeSystemDefinitionNode,
+  Kind,
+} from 'graphql';
+import { TransformerPluginProvider, TransformerPluginType } from '@aws-amplify/graphql-transformer-interfaces';
+
+export function makeSeenTransformationKey(
+  directive: DirectiveNode,
+  type: TypeDefinitionNode,
+  field?: FieldDefinitionNode | InputValueDefinitionNode | EnumValueDefinitionNode,
+  arg?: InputValueDefinitionNode,
+  index?: number,
+): string {
+  let key = '';
+  if (directive && type && field && arg) {
```

Comment:
if this is true, then the below if condition is also true which will overwrite this assignment

Reply:
nvm just realized it's an else if. However, I'd move the else if up one line so it doesn't look like a new if statement:
```
if {

} else if {

}
```